internal/throttling/memory_gorate.go

```diff
@@ -0,0 +1,50 @@
+package throttling
+
+import (
+	"sync"
+	"time"
+
+	gorate "golang.org/x/time/rate"
+
+	"github.com/rudderlabs/rudder-server/internal/cachettl"
+)
+
+type goRate struct {
+	mu    sync.Mutex
+	store *cachettl.Cache
+}
+
+func (r *goRate) limit(key string, cost, rate, periodInSecs int64) *goRateReservation {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	if r.store == nil {
+		r.store = cachettl.New()
+	}
+
+	window := time.Duration(periodInSecs) * time.Second
+	l, ok := r.store.Get(key).(*gorate.Limiter)
```

Comment:
when a limiter is used, it should update (refresh) its ttl in the cache so that it only expires due to inactivity